### ISSUE TITLE
Universal way to reset documents: b.revert()

### ISSUE THREAD

**Issue by trych** [basiljs/basil.js#133]
I think I now found a good solution to a problem that has been bugging me since July and that I have already described in #78.

While developing a script it is of great help to put `b.clear(b.doc());` at the beginning of the script, so the script starts with a cleared document each round. However, if the script that is developed contains steps like creating new pages, layers, colors etc. this method fails, as these are not cleared by `b.clear()`. So in this case you either have to do a manual undo after each execution or you have to close the document and open a new one, which you could also do with `b.close(false);`.

`b.close(false);` however would not work on documents that I have saved before executing (for example when I need to use basil's data folder). So I have to manually undo the script or revert to the last save.

Therefore I propose a function `b.revert()` that does exactly that, to have one universal command to always reset the document to it's initial state (as InDesign's `Revert` command does). So, if I have my document saved, it returns to the last saved state. If I have my document not saved yet *it would close the unsaved document without saving and open a new empty document*, kind of "reverting to the initial empty document".

That way I could always just put a `b.revert()` command at the top of my script during development and not have to worry if it is a script with a saved doc, a script that adds layers or pages or a script that is not saved yet (and therefore always starts from a new empty document).

Thoughts? 😊

**Comment by fabianmoronzirfas** [comment]
Sounds great. Way better then using `b.clear` with additional parameters

**Comment by trych** [comment]
Bonus: it would also be much faster than `b.clear(b.doc())` and you could very nicely and quickly create 100s of custom PDFs from scratch like so:

```js
for(var i = 0; i < 100; i += 1) {
  b.revert();
  // do some stuff, maybe even add random colors or layers or stuff
  b.savePDF(i + ".pdf", false);
}
```

**Comment by b-g** [comment]
Sound great +1!

But should it not be `b.revert(b.doc())`? That way you have at least a hint what is reverted ...

**Comment by trych** [comment]
> But should it not be b.revert(b.doc())?

I think as long as there are not any options of what is going to be reverted there should be no argument. As a doc is the only thing that *can* be reverted, I would leave it without argument. This would be also consistent with functions like `b.close()`, `b.size()` etc.

**Comment by b-g** [comment]
@trych OK + convinced! Just `b.revert()` is better in terms of consistency.

**Comment by fabianmoronzirfas** [comment]
Should `b.revert()` return the current doc?  
e.g. When using revert on an unsaved document. Wouldn't it be nice to directly get the new doc from revert?

**Comment by trych** [comment]
Sure, no harm in doing that.

**Comment by trych** [comment]
Closed via #204.
